lib/stdlib/src/supervisor.erl

```diff
@@ -1509,18 +1509,13 @@ add_restart(State) ->
 	    {terminate, State1}
     end.
 
-add_restart([R|Restarts], Now, Period) ->
-    case inPeriod(R, Now, Period) of
-	true ->
-	    [R|add_restart(Restarts, Now, Period)];
-	_ ->
-	    []
-    end;
-add_restart([], _, _) ->
-    [].
-
-inPeriod(Then, Now, Period) ->
-    Now =< Then + Period.
+add_restart(Restarts0, Now, Period) ->
+    Treshold = Now - Period,
+    Restarts1 = lists:takewhile(
```

Comment:
Just wondering if makes sense to replace `lists:takewhile/2` to a list comprehensions:
```erlang
...
Restarts1 = [R || R <- Restarts0, R >= Treshold],
...
```

Reply:
The restarts in the list are naturally ordered newest to oldest, so we may stop processing early when we hit the first out-of-period timestamp, because anything beyond it will be even older. This is just what `lists:takewhile/2` does. A list comprehension would traverse the list in full, like `lists:filter/2` would.